src/box/lua/trigger.h

```diff
@@ -9,12 +9,55 @@
 extern "C" {
 #endif /* defined(__cplusplus) */
 
+struct event;
+
 /**
  * Initializes module trigger.
  */
 void
 box_lua_trigger_init(struct lua_State *L);
 
+/**
+ * Creates a Lua trigger, replaces an existing one, or deletes a trigger.
+ *
+ * The function accepts a Lua stack. Values starting from index bottom are
+ * considered as the function arguments.
+ *
+ * The function supports two API versions.
+ *
+ * The first version - key-value arguments. The function is called with one Lua
```

Comment:
I don't understand why we need to extend the legacy API. AFAIU we're planning to drop it sooner or later so we need this solely as a backward compatibility. Also, I think that we should print a warning on an attempt to use it. I'll talk to PM to clarify these points.

Reply:
#8659 is about it. We decided that there are users who wants to use named triggers, but does not want to completely rewrite lots of code that sets triggers - they'll just need to add a third argument. And key-value API won't be deprecated.

What's about deprecation message, I was going to add it as a separate patch after we will deal with all triggers (`net.box`, transactional ones, etc.).

Reply:
Talked to @unera. He said that we *will* drop the old API completely in 4.0, including the extension introduced in this PR. He also said that he only asked to add the third argument to `trigger.reset` and not the table argument (point 2 in #8659).

Talked to @Totktonada. He said that he doesn't remember discussion of the table argument to `trigger.reset` (point 2 in #8659). Since we're planning to drop it, he says he wouldn't add it.

Something seems to be broken in the feature design process... Since we already implemented it, let's leave it as is.